test/unit/projection/test_quad.cpp

```diff
@@ -0,0 +1,118 @@
+#include <StdRegions/StdExpansion.h>
+#include <StdRegions/StdMatrixKey.h>
+#include <StdRegions/StdQuadExp.h>
+
+#include <gtest/gtest.h>
+#include <nektar_interface/projection/algorithm_types.hpp>
+#include <nektar_interface/projection/device_data.hpp>
+
+#include <nektar_interface/projection/auto_switch.hpp>
+#include <nektar_interface/projection/quad.hpp>
+#include <sycl/sycl.hpp>
+
+#include "create_data.hpp"
+
+using namespace NESO::Project;
+using namespace Nektar;
+using namespace Nektar::LibUtilities;
+using namespace Nektar::StdRegions;
+
+class ProjectQuadCell : public ::testing::TestWithParam<TestData2D> {
+public:
+  double Integrate(TestData2D &test_data) {
+    sycl::queue Q{sycl::default_selector_v};
+    size_t const ndof = test_data.ndof;
+    size_t const nmode = ndof - 1;
+    PointsKey pk{ndof, eGaussLobattoLegendre};
+    BasisKey bk{eModified_A, nmode, pk};
+    StdExpansion *Shape = new StdQuadExp{bk, bk};
+
+    auto [data, pntrs] =
+        create_data(Q, nmode * nmode, test_data.val, test_data.x, test_data.y);
+    std::optional<sycl::event> event;
+    AUTO_SWITCH(static_cast<int>(nmode), event, ThreadPerCell::template project,
+                FUNCTION_ARGS(data, 0, Q), double, 1, 1, eQuad<ThreadPerCell>);
+    if (event) {
+      event.value().wait();
+    }
+    auto buffer = std::vector<double>(Shape->GetNcoeffs(), double(0.0));
+    Q.memcpy(buffer.data(), data.dofs, Shape->GetNcoeffs() * sizeof(double))
+        .wait();
+    Array<OneD, double> phi(Shape->GetNcoeffs(), buffer.data());
+    Array<OneD, double> coeffs(Shape->GetNcoeffs());
+
+    // Multiply by inverse mass matrix
+    StdMatrixKey masskey(eInvMass, Shape->DetShapeType(), *Shape);
+    DNekMatSharedPtr matsys = Shape->GetStdMatrix(masskey);
+    NekVector<NekDouble> coeffsVec(Shape->GetNcoeffs(), coeffs, eWrapper);
+    NekVector<NekDouble> phiVec(Shape->GetNcoeffs(), phi, eWrapper);
+    coeffsVec = (*matsys) * phiVec;
+
+    free_data(Q, pntrs);
+    // Transfrom to physical space
+    Array<OneD, double> phys(Shape->GetTotPoints());
+    Shape->BwdTrans(coeffs, phys);
+    return Shape->Integral(phys);
+  }
+};
+
+TEST_P(ProjectQuadCell, IntegralIsRight) {
+  auto test_data = GetParam();
+  ASSERT_NEAR(test_data.val, Integrate(test_data), 1.0e-6);
```

Comment:
Personally I would add a test where you use Nektar functions to compute the RHS of the linear system (i.e. use nektar to compute the RHS values by evaluating the basis functions at the particle positions) then check your computation matches.

Reply:
Is this what your tests already do?

Reply:
Yes, here for example: https://github.com/ExCALIBUR-NEPTUNE/NESO/blob/c13c49e8b1d72b56cc92e949b78304cb7ad3c9d5/test/unit/nektar_interface/test_particle_function_projection.cpp#L371

Reply:
As the new functions are still tested by your existing tests I mark this as resolved

Reply:
I can't switch in my tests because  the call is too low level it doesn't go via the `FunctionProjectBasis` method.
If we wanted to add new tests or adapt the original ones -  I have enabled the runtime  switch via an environment variable  you would have to call `setenv` in the test but it is possible.
But I don't really want to write tests that depend on code I want to delete eventually. In the end the new versions are still tested with your old tests + my new tests.